pkg/network/server.go

```diff
@@ -277,6 +277,19 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	if err := p.WriteMsg(NewMessage(s.Net, CMDPing, ping)); err != nil {
+		return err
+	}
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.HeaderHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, pong))
```

Comment:
And on the `pong` side there should be some outbound ping request from us for it to be successful (spurious pongs from the other side are not OK). I don't think there is a need to send anything to the peer in response to `pong`.